File: frontend/src/api/order/dto/OrderDto.ts
import { Address } from "../../common/dto/Address";
import { OrderStatus } from "../enums/OrderStatus";
import { OrderItemDto } from "./OrderItemDto";
import { PaymentInfo } from "./PaymentInfo";

export interface OrderDto {
  id: string; // ObjectId as a string
  userId: string;
  deliveryAddress: Address; // Assuming Address is a string
  status: OrderStatus;
  promoCode: string;
  paymentInfo: PaymentInfo; // Assuming PaymentInfo is a string
  orderedAt: string; // ISO date-time string
  readyAt: string; // ISO date-time string
  completedAt: string; // ISO date-time string
  items: OrderItemDto[];
}

File: services/order-service/src/main/java/com/mealmap/orderservice/core/dto/order/OrderDto.java
package com.mealmap.orderservice.core.dto.order;

import com.mealmap.orderservice.core.enums.OrderStatus;
import com.mealmap.orderservice.doc.value.Address;
import com.mealmap.orderservice.doc.value.PaymentInfo;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import org.bson.types.ObjectId;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@Schema(description = "Информация о заказе")
public class OrderDto {
    private ObjectId id;

    private String userId;

    private Address deliveryAddress;

    private OrderStatus status;

    private String promoCode;

    private PaymentInfo paymentInfo;

    private LocalDateTime orderedAt;

    private LocalDateTime readyAt;

    private LocalDateTime completedAt;

    private List<OrderItemDto> items;
}
